jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -917,227 +919,497 @@ ecma_number_to_int32 (ecma_number_t num) /**< ecma-number */
+
+#define GRISU3_SIGN_BIT                 0x8000000000000000ULL
+#define GRISU3_EXPONENT_MASK            0x7FF0000000000000ULL
+#define GRISU3_FRACTION_MASK            0x000FFFFFFFFFFFFFULL
+#define GRISU3_IMPLICIT_ONE             0x0010000000000000ULL
+#define GRISU3_EXPONENT_POSITION        52
+#define GRISU3_EXPONENT_BIAS            1075
+#define GRISU3_1_LOG2_10                0.30102999566398114 /* 1 / lg(10) */
+#define GRISU3_MIN_TARGET_EXP           -60
+#define GRISU3_32_BIT_MASK              0xFFFFFFFFULL
+#define GRISU3_ENCODED_FLOAT_FRACT_SIZE 64
+#define GRISU3_MIN_CACHED_EXP           -348
+#define GRISU3_CACHED_EXP_STEP          8
+
 /**
-  * Perform conversion of 128-bit binary representation of number
-  * to decimal representation with decimal exponent.
-  */
-static void
-ecma_number_helper_binary_to_decimal (ECMA_NUMBER_CONVERSION_128BIT_INTEGER_ARG (fraction_uint128), /**< mantissa */
-                                      int32_t binary_exponent, /**< binary exponent */
-                                      int32_t *out_decimal_exp_p) /**< [out] decimal exponent */
+ * Encoded floating point type
+ */
+typedef struct
 {
-  int32_t decimal_exp = 0;
+  uint64_t fract;
+  int exp;
+} encoded_float_t;
 
-  if (binary_exponent > 0)
-  {
-    while (binary_exponent > 0)
-    {
-      if (!ECMA_NUMBER_CONVERSION_128BIT_INTEGER_IS_HIGH_BIT_MASK_ZERO (fraction_uint128, 124))
-      {
-        ECMA_NUMBER_CONVERSION_128BIT_INTEGER_INC (fraction_uint128);
-        ECMA_NUMBER_CONVERSION_128BIT_INTEGER_RIGHT_SHIFT (fraction_uint128);
-        binary_exponent++;
-      }
-      else
-      {
-        ECMA_NUMBER_CONVERSION_128BIT_INTEGER (fraction_uint128_tmp);
-        ECMA_NUMBER_CONVERSION_128BIT_INTEGER_COPY (fraction_uint128_tmp, fraction_uint128);
-        ECMA_NUMBER_CONVERSION_128BIT_INTEGER_DIV_10 (fraction_uint128_tmp);
-        ECMA_NUMBER_CONVERSION_128BIT_INTEGER_MUL_10 (fraction_uint128_tmp);
+/**
+ * Power type
+ */
+typedef struct
+{
+  uint64_t fract;
+  int16_t b_exp;
+  int16_t d_exp;
+} power_t;
 
-        if (!ECMA_NUMBER_CONVERSION_128BIT_INTEGER_ARE_EQUAL (fraction_uint128, fraction_uint128_tmp)
-            && ECMA_NUMBER_CONVERSION_128BIT_INTEGER_IS_HIGH_BIT_MASK_ZERO (fraction_uint128, 123))
-        {
-          ECMA_NUMBER_CONVERSION_128BIT_INTEGER_LEFT_SHIFT (fraction_uint128);
-          binary_exponent--;
-        }
-        else
-        {
-          ECMA_NUMBER_CONVERSION_128BIT_INTEGER_DIV_10 (fraction_uint128);
-          decimal_exp++;
-        }
-      }
-    }
-  }
-  else if (binary_exponent < 0)
-  {
-    while (binary_exponent < 0)
-    {
-      if (ECMA_NUMBER_CONVERSION_128BIT_INTEGER_IS_LOW_BIT_MASK_ZERO (fraction_uint128, 0)
-          || !ECMA_NUMBER_CONVERSION_128BIT_INTEGER_IS_HIGH_BIT_MASK_ZERO (fraction_uint128, 124))
-      {
-        ECMA_NUMBER_CONVERSION_128BIT_INTEGER_RIGHT_SHIFT (fraction_uint128);
+#define POWER_CACHE_ITEM(fract, b_exp, d_exp) \
+  { fract, b_exp, d_exp }
```

Comment:
this is a one line initializer, it is acceptable. There are several examples in the current code base.
Is this what you are thinking of?

```C++
{
  fract,
  b_exp,
  d_exp
}
```